File: src/main/java/com/revhire/service/SavedJobService.java
package com.revhire.service;

import com.revhire.model.SavedJob;
import java.util.List;

public interface SavedJobService {
    boolean saveJob(int userId, int jobId);

    boolean unsaveJob(int userId, int jobId);

    boolean isJobSaved(int userId, int jobId);

    List<SavedJob> getSavedJobs(int userId);
}

File: revhire-frontend/src/app/core/services/saved-job.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';
import { Observable } from 'rxjs';
import { SavedJob } from '../../models/saved-job.model';

@Injectable({
    providedIn: 'root'
})
export class SavedJobService {
    private apiUrl = `${environment.apiUrl}/saved-jobs`;

    constructor(private http: HttpClient) { }

    getSavedJobs(): Observable<SavedJob[]> {
        return this.http.get<SavedJob[]>(`${this.apiUrl}`);
    }

    saveJob(jobId: number): Observable<any> {
        return this.http.post(`${this.apiUrl}/${jobId}`, {});
    }

    unsaveJob(jobId: number): Observable<any> {
        return this.http.delete(`${this.apiUrl}/${jobId}`);
    }
}
